File: Math/CPP/x_of_a_kind_in_card_deck.cpp
#include <numeric>
#include <map>
#include <vector>
using namespace std;

class Solution {
public:
    bool hasGroupsSizeX(vector<int>& deck) {
        map<int, int> counter;
        int res;
        for (int n: deck) counter[n] += 1;
        for (auto i : counter) res = __gcd(i.second, res);
        return res > 1;
    }
};

File: Math/Python/x_of_a_kind_in_card_deck.py
import math

from collections import Counter
from functools import reduce

class Solution:
    def hasGroupsSizeX(self, deck: list[int]) -> bool:
        def gcd(a, b):
            while b: a, b = b, a % b
            return a
        cnt = Counter(deck).values()
        return reduce(gcd, cnt) > 1
    
    # one line
    def hasGroupsSizeX(self, deck: list[int]) -> bool:
        return reduce(math.gcd, Counter(deck).values()) > 1